Python/pystate.c

```diff
@@ -1064,6 +1066,73 @@ _PyInterpreterState_FailIfRunningMain(PyInterpreterState *interp)
 // accessors
 //----------
 
+PyObject *
+PyUnstable_InterpreterState_GetMainModule(PyInterpreterState *interp)
+{
+    PyObject *modules = _PyImport_GetModules(interp);
+    if (modules == NULL) {
+        PyErr_SetString(PyExc_RuntimeError, "interpreter not initialized");
+        return NULL;
+    }
+    return PyMapping_GetItemString(modules, "__main__");
+}
+
+PyObject *
+PyInterpreterState_GetDict(PyInterpreterState *interp)
```

Comment:
Why is `PyUnstable_InterpreterState_GetMainModule()` unstable but not this one?

Reply:
I'm not sure.  I'll take a look.

Reply:
(The history of both functions is unrelated to this PR.  Here I'm just adjusting where they are grouped in the file.)

`PyInterpreterState_GetDict()` was added in 2019 at the request of users (PyPy, IIRC) and started out as public API.

`PyUnstable_InterpreterState_GetMainModule()` started out as internal API and I moved it (tentatively) to the public API in July of last year.  I marked it as unstable because I wasn't completely sure we wanted to keep it in the public API.